src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -502,6 +505,87 @@ public OrganisationEntityResponseV2 retrieveOrganisationForV2Api(
         return new OrganisationEntityResponseV2(organisation, true, isPendingPbaRequired, false, true);
     }
 
+    @Override
+    public MultipleOrganisationsResponse retrieveOrganisationsByProfileIds(List<String> organisationProfileIds,
+                                                                           UUID searchAfter) {
+        List<String> orgTypes = convertProfileIdsToOrgTypes(organisationProfileIds);
+
+        boolean orgIdFilterProvided = !orgTypes.isEmpty();
+        boolean searchAfterProvided = searchAfter != null;
+
+        List<Organisation> organisations = new ArrayList<>();
+
+        if (orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeInAndIdGreaterThan(orgTypes, searchAfter);
+        }
+
+        if (orgIdFilterProvided && !searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeIn(orgTypes);
+        }
+
+        if (!orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByIdGreaterThan(searchAfter);
+        }
+
+        if (!orgIdFilterProvided && !searchAfterProvided) {
+            organisations = organisationRepository.findAll();
+        }
+
+        return new MultipleOrganisationsResponse(organisations, false);
+    }
+
+    @Override
+    public MultipleOrganisationsResponse retrieveOrganisationsByProfileIdsWithPageable(
+            List<String> organisationProfileIds,
+            Integer pageSize,
+            UUID searchAfter) {
+        List<String> orgTypes = convertProfileIdsToOrgTypes(organisationProfileIds);
+
+        Pageable pageableObject = createPageableObject(0, pageSize, Sort.by(Sort.DEFAULT_DIRECTION, "id"));
+        boolean orgIdFilterProvided = organisationProfileIds != null && !organisationProfileIds.isEmpty();
```

Comment:
boolean orgIdFilterProvided = !orgTypes.isEmpty();
 Why can't we use this

Reply:
done